File: Problems.CPP/Problem_8.cpp
# include <iostream>

using namespace std;

class  Problem_8
{
public:

	int ReadPositiveNumber(string message)
	{
		int number;

		do
		{
			cout << message;
			cin >> number;

		} while (number < 0);

			return number;
	}

	int checkDigitFrequncy(int number, int numberToCheck)
	{
		int FreqCount = 0;
		int remainder = 0;

		while (number > 0)
		{
			remainder = number % 10;
			number = number / 10;

			if (numberToCheck == remainder)
			{
				FreqCount++;
			}

		}

		return FreqCount;
	}
};

File: Problems.CSharp/Problems/0008-Count digit frequency/Problem_8.cs
namespace Problems.CSharp.Problems._0008_Count_digit_frequency
{
    public class Problem_8
    {
        public int ReadPositiveNumber(string Message)
        {
            int number;

            do
            {
                Console.Write(Message);
            } while (!int.TryParse(Console.ReadLine(), out number) || number <= 0);

            return number;
        }

        public int CheckDigitFrequency(int number, int numberToCheck)
        {
            int freqCount = 0;
            int remainder = 0;

            while(number > 0)
            {
                remainder = number % 10;
                number = number / 10;

                if(numberToCheck == remainder)
                {
                    freqCount++;
                }
            }

            return freqCount;
        }
    }
}
